lnbits/core/views/generic.py

```diff
@@ -166,18 +166,25 @@ async def extensions_install(
 async def wallet(
     request: Request,
     usr: UUID4 = Query(...),
-    wal: UUID4 = Query(...),
+    wal: Optional[UUID4] = Query(None),
 ):
     user_id = usr.hex
-    wallet_id = wal.hex
-
     user = await get_user(user_id)
 
     if not user:
         return template_renderer().TemplateResponse(
             "error.html", {"request": request, "err": "User does not exist."}
         )
 
+    if not wal:
+        if len(user.wallets) == 0:
+            return template_renderer().TemplateResponse(
+                "error.html", {"request": request, "err": "User has no wallets."}
+            )
+        return RedirectResponse(url=f"/wallet?usr={user_id}&wal={user.wallets[0].id}")
```

Comment:
- `user.wallets[0]` can now be `None` as the `user always has a wallet` rule is no longer valid

Reply:
i catch that about with `len(user.wallets) == 0`